docs/sources/tempo/traceql/construct-traceql-queries.md

```diff
@@ -0,0 +1,609 @@
+
+An unanchored query, such as: { span.foo =~ "bar" } is now treated as: { span.foo =~ "^bar$" }.
+
+If you use TraceQL with regular expressions in your Grafana dashboards and you want the unanchored behavior, update the queries to use the unanchored version, such as { span.foo =~ ".*bar.*"}.
+
+For example, to find all traces where an `http.status_code` attribute in a span are greater than `400` but less than equal to `500`:
+
+```
+{ span.http.status_code >= 400 && span.http.status_code < 500 }
+```
+
+This works for `http.status_code` values that are strings as well using lexographic ordering:
+
+```
+{ span.http.status_code >= "400" }
+```
+
+Find all traces where the `http.method` attribute is either `GET` or `DELETE`:
+
+```
+{ span.http.method =~ "DELETE|GET" }
+```
+
+Find all traces where `any_attribute` is not `nil` or where `any_attribute` exists in a span
+```
+{ .any_attribute != nil }
+```
+
+### Field expressions
+
+Fields can also be combined in various ways to allow more flexible search criteria.
+A field expression is a composite of multiple fields that define all of the criteria that must be matched to return results.
+
+#### Examples
+
+Find traces with `success` `http.status_code` codes:
+
+```
+{ span.http.status_code >= 200 && span.http.status_code < 300 }
+```
+
+Find traces where a `DELETE` HTTP method was used and the intrinsic span status wasn't OK:
+
+```
+{ span.http.method = "DELETE" && status != ok }
+```
+
+Both expressions require all conditions to be true on the same span.
+The entire expression inside of a pair of `{}` must be evaluated as true on a single span for it to be included in the result set.
+
+In the above example, if a span includes an `.http.method` attribute set to `DELETE` where the span also includes a `status` attribute set to `ok`, the trace would not be included in the returned results.
+
+## Combine spansets using operators
+
+Spanset operators let you select different sets of spans from a trace and then make a determination between them.
+
+### Logical
+
+These spanset operators perform logical checks between the sets of spans.
+
+- `{condA} && {condB}` - The and operator (`&&`) checks that both conditions found matches.
+- `{condA} || {condB}` - The union operator (`||`) checks that either condition found matches. This functions as an "OR" statement.
+
+For example, to find a trace that went through two specific `cloud.region`:
+
+```
+{ resource.cloud.region = "us-east-1" } && { resource.cloud.region = "us-west-1" }
+```
+
+Note the difference between the previous example and this one:
+
+```
+{ resource.cloud.region = "us-east-1" && resource.cloud.region = "us-west-1" }
+```
+
+The second expression returns no traces because it's impossible for a single span to have a `resource.cloud.region` attribute that's set to both region values at the same time.
+
+You can use a similar query to find a trace that passed through either `us-east-1` or `us-west-1` cloud regions:
```

Comment:
OR example, per request in https://github.com/grafana/tempo/issues/4138#issuecomment-2905410873

Reply:
I've. added a clarification.